core/src/UserSettingsDialog.html

```diff
@@ -0,0 +1,289 @@
+<script>
+    import {
+        createEventDispatcher,
+        onMount,
+        onDestroy,
+        beforeUpdate,
+        getContext
+    } from 'svelte';
+    import { NavigationHelpers } from './utilities/helpers';
+    import UserSettingsEditor from './UserSettingsEditor.html';
+    import { CSS_BREAKPOINTS } from './utilities/constants';
+    import { LuigiConfig } from './core-api';
+    import { TOP_NAV_DEFAULTS } from './utilities/luigi-config-defaults';
+    export let schemaObj;
+    export let userSettingsData;
+    const dispatch = createEventDispatcher();
+    let userSettingGroup;
+    let storedUserSettings = {};
+    let userSettingsGroupKey;
+    let userSettingsSchema;
+    let displayEditor = false;
+    let previousUserSettings = {};
+    let getTranslation = getContext('getTranslation');
+    let dialogHeader = TOP_NAV_DEFAULTS.userSettingsDialog.dialogHeader;
+    let saveBtn = TOP_NAV_DEFAULTS.userSettingsDialog.saveBtn;
+    let dismissBtn = TOP_NAV_DEFAULTS.userSettingsDialog.dismissBtn;
+
+    onMount(() => {
+        const usLocalization = LuigiConfig.getConfigValue('settings.userSettings.userSettingsDialog');
+        if (usLocalization) {
+            dialogHeader = usLocalization.dialogHeader ? usLocalization.dialogHeader : dialogHeader;
+            saveBtn = usLocalization.saveBtn ? usLocalization.saveBtn : saveBtn;
+            dismissBtn = usLocalization.dismissBtn ? usLocalization.dismissBtn : dismissBtn;
+        }
+        let key;
+        for (let k in userSettingsData[0]) {
+            key = k;
+            break;
+        }
+        LuigiConfig.readUserSettings().then(userSettingsFromStorage => {
+            previousUserSettings = JSON.parse(JSON.stringify(userSettingsFromStorage));
+            if (userSettingsFromStorage === null) {
+                storedUserSettings;
+            } else {
+                storedUserSettings = userSettingsFromStorage;
+            }
+            openEditor([key, userSettingsData[0][key]]);
+        });
```

Comment:
It is pretty likely that an application would perform external call to read the setting; would be usefull to also implement the method catch in the Promise in case the readUserSetting is generating any error? We could display it with an Alert

Reply:
👍 Good point. I will add it.